test/components/views/right_panel/UserInfo-test.tsx

```diff
@@ -43,147 +64,927 @@ jest.mock("../../../../src/utils/DMRoomMap", () => {
     };
 });
 
-describe("<UserInfo />", () => {
-    const defaultUserId = "@test:test";
-    const defaultUser = new User(defaultUserId);
-
-    const mockClient = mocked({
-        getUser: jest.fn(),
-        isGuest: jest.fn().mockReturnValue(false),
-        isUserIgnored: jest.fn(),
-        isCryptoEnabled: jest.fn(),
-        getUserId: jest.fn(),
+const mockRoom = mocked({
+    roomId: "!fkfk",
+    getType: jest.fn().mockReturnValue(undefined),
+    isSpaceRoom: jest.fn().mockReturnValue(false),
+    getMember: jest.fn().mockReturnValue(undefined),
+    getMxcAvatarUrl: jest.fn().mockReturnValue("mock-avatar-url"),
+    name: "test room",
+    on: jest.fn(),
+    currentState: {
+        getStateEvents: jest.fn(),
         on: jest.fn(),
-        isSynapseAdministrator: jest.fn().mockResolvedValue(false),
-        isRoomEncrypted: jest.fn().mockReturnValue(false),
-        doesServerSupportUnstableFeature: jest.fn().mockReturnValue(false),
-        mxcUrlToHttp: jest.fn().mockReturnValue("mock-mxcUrlToHttp"),
-        removeListener: jest.fn(),
-        currentState: {
-            on: jest.fn(),
-        },
-    } as unknown as MatrixClient);
+    },
+    getEventReadUpTo: jest.fn(),
+} as unknown as Room);
+
+const mockClient = mocked({
+    getUser: jest.fn(),
+    isGuest: jest.fn().mockReturnValue(false),
+    isUserIgnored: jest.fn(),
+    isCryptoEnabled: jest.fn(),
+    getUserId: jest.fn(),
+    on: jest.fn(),
+    isSynapseAdministrator: jest.fn().mockResolvedValue(false),
+    isRoomEncrypted: jest.fn().mockReturnValue(false),
+    doesServerSupportUnstableFeature: jest.fn().mockReturnValue(false),
+    mxcUrlToHttp: jest.fn().mockReturnValue("mock-mxcUrlToHttp"),
+    removeListener: jest.fn(),
+    currentState: {
+        on: jest.fn(),
+    },
+    checkDeviceTrust: jest.fn(),
+    checkUserTrust: jest.fn(),
+    getRoom: jest.fn(),
+    credentials: {},
+    setPowerLevel: jest.fn(),
+} as unknown as MatrixClient);
+
+const defaultUserId = "@test:test";
+const defaultUser = new User(defaultUserId);
+
+beforeAll(() => {
+    jest.spyOn(MatrixClientPeg, "get").mockReturnValue(mockClient);
+});
 
+afterEach(() => {
+    mockClient.getUser.mockClear().mockReturnValue({} as unknown as User);
+    jest.spyOn(MatrixClientPeg, "get").mockReturnValue(mockClient);
```

Comment:
shouldn't we do this `jest.spyOn(MatrixClientPeg, "get").mockReturnValue(mockClient);` in a `beforeEach`, which would save having to have a separate `beforeAll`?

Reply:
_much_ tidier, thanks